apps/examples/rtc/rtc_main.c

```diff
@@ -150,10 +161,230 @@ static void *rtc_test(void *arg)
+				 */
+				g_relset = false;
+				break;
+			case 'r':
+				/* -r: alarm is set by relative time (RTC_SET_RELATIVE ioctl call).
+				 */
+				g_relset = true;
+				break;
+			case 'c':
+				cancelmode = true;
+				break;
+			default:
+				fprintf(stderr, "<unknown parameter '-%c'>\n\n", opt);
+				/* fall through */
+
+			case '?':
+			case ':':
+				badarg = true;
+		}
+	}
+
+	if (badarg) {
+		show_usage(argv[0]);
+		return EXIT_FAILURE;
+	}
+
+	if (!cancelmode) {
+		/* Get the number of seconds until the alarm expiration */
+		if(g_relset) {
+			seconds = strtoul(argv[2], NULL, 10);
+			if (seconds < 1) {
+				fprintf(stderr, "ERROR: Invalid number of seconds: %lu\n", seconds);
+				show_usage(argv[0]);
+				return EXIT_FAILURE;
+			}
+		} else {
+			sscanf(argv[2], "%d:%d:%d", &hrs, &mins, &seconds);
+		}
+	}
+
+	/* Open the RTC driver */
+
+	printf("Opening %s\n", RTC_DEVPATH);
+
+	fd = open(RTC_DEVPATH, O_RDWR);
+	if (fd < 0) {
+		fprintf(stderr, "ERROR: Failed to open %s: %d\n", RTC_DEVPATH, errno);
+		return EXIT_FAILURE;
+	}
+
+	if (cancelmode) {
+		ret = ioctl(fd, RTC_CANCEL_ALARM, (unsigned long)RTC_ALARM_ID);
+		if (ret < 0) {
+			fprintf(stderr, "ERROR: RTC_CANCEL_ALARM ioctl failed: %d\n", errno);
+			close(fd);
+			return EXIT_FAILURE;
+		}
+
+		printf("Alarm has been canceled\n");
+	} else { /*!cancelmode */
+
+		sigemptyset(&set);
+		sigaddset(&set, RTC_ALARM_SIGNO);
+		ret = sigprocmask(SIG_UNBLOCK, &set, NULL);
+		if (ret != OK) {
+			fprintf(stderr, "ERROR: sigprocmask failed: %d\n", errno);
+			close(fd);
+			return EXIT_FAILURE;
+		}
+
+		act.sa_sigaction = alarm_handler;
+		act.sa_flags = SA_SIGINFO;
+		sigfillset(&act.sa_mask);
+		sigdelset(&act.sa_mask, RTC_ALARM_SIGNO);
+		ret = sigaction(RTC_ALARM_SIGNO, &act, NULL);
+		if (ret < 0) {
+			fprintf(stderr, "ERROR: sigaction failed: %d\n", errno);
+			close(fd);
+			return EXIT_FAILURE;
+		}
+
+		/* Set the alarm */
+		if (g_relset) {
+
+			setrel.id = RTC_ALARM_ID;
+			setrel.pid = 0;
+			setrel.reltime = (time_t)seconds;
+			setrel.signo = RTC_ALARM_SIGNO;
+			setrel.sigvalue.sival_int = RTC_ALARM_ID;
+			ret = ioctl(fd, RTC_SET_RELATIVE, (unsigned long)((uintptr_t)&setrel));
+			if (ret < 0) {
+				fprintf(stderr, "ERROR: RTC_SET_RELATIVE ioctl failed: %d\n", errno);
+				close(fd);
+				return EXIT_FAILURE;
+			}
+			ret = ioctl(fd, RTC_RD_TIME, (unsigned long)&g_alarm_time);
+			if (ret < 0) {
+				fprintf(stderr, "ERROR: RTC_RD_TIME ioctl failed: %d\n", errno);
+				close(fd);
+				return EXIT_FAILURE;
+			}
+			printf("Alarm is set to expire in %lu seconds\n", seconds);
+			sleep(seconds + 1);
+		} else {
+
+			setalarm.id = RTC_ALARM_ID;
+			setalarm.pid = 0;
+
+			ret = ioctl(fd, RTC_RD_TIME, (unsigned long)&g_alarm_time);
+			if (ret < 0) {
+				fprintf(stderr, "ERROR: RTC_RD_TIME ioctl failed: %d\n", errno);
+				close(fd);
+				return EXIT_FAILURE;
+			}
+			user_time = hrs * 3600 + mins * 60 + seconds;
```

Comment:
What's the meaning of the `user_time` and `cur_time`?

Reply:
user_time: time specified by user to set alarm
cur_time: Current time